### PR TITLE
Add `hasMonth` and `hasMonthValue` to `LocalDateTime` assertions

### LINKED ISSUE #2541: Add more basic assertions for Local(Date|Time)
#### Summary

It would be nice if `AbstractLocal(Date|Time)Assert` has basic assertions for field values of `actual`.

LocalDate
- [x] `hasYear` - 9dc086d82aa58a755f36cf88449b0ca74288bb63
- [x] `hasMonth` / `hasMonthValue` - 40c979c253571743af8d68f18921a061e10cfcec
- [x] `hasDayOfMonth` - 253194d00395e1568e1be752b0075c2033672f97

LocalTime
- [x] `hasHour` - 8c54e417b697314e55170668bd96c236fa3c96d6
- [x] `hasMinute` - ec1fcaefb767d0ed18a7e6ea48779c78635d1399
- [x] `hasSecond` - 6a4e54cbad5198246ae41a21d8432c503b5c70ce
- [x] `hasNano` - b756459

LocalDateTime
- [x] `hasYear` - 710177b
- [x] `hasMonth` / `hasMonthValue`  - d0c87895c2a052473e8c4a7800fa7fee0a20dccb
- [x] `hasDayOfMonth` - 710177b
- [x] `hasHour` - 7cbe45c42f0a31bc5993a8352cf5e0d4feda0fae
- [ ] `hasMinute` - @Leo0506
- [x] `hasSecond` - 99e4f652cecc6f4492a0aec7bcfc7e97518a032a
- [x] `hasNano` -  76012bb

#### Examples

```java
  final LocalDate actual = LocalDate.now();
  assertThat(actual).hasYear(actual.getYear());
  assertThat(actual).hasMonth(actual.getMonth());
  assertThat(actual).hasMonthValue(actual.getMonthValue());
  assertThat(actual).hasDayOfMonth(actual.getDayOfMonth());

  final LocalTime actual = LocalTime.now();
  assertThat(actual).hasHour(actual.getHour());
  assertThat(actual).hasMinute(actual.getMinute());
  assertThat(actual).hasSecond(actual.getSecond());
  assertThat(actual).hasNano(actual.getNano());
```

### PR DESCRIPTION
**Pull request by Zhou-Yicheng**
#### Check List:
* Fixes partially #2541
* Unit tests : YES
* Javadoc with a code example (on API only) : YES
* PR meets the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Zhou-Yicheng** [comment]
Maybe the failure is related to #2040

File: src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_hasMonthValue_Test.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    then(assertionError).hasMessage(shouldHaveDateField(actual, "month", monthVal).create());
+  }
+
+  @Test
+  void should_pass_if_actual_has_same_month_value() {
+    // GIVEN
+    LocalDateTime actual = LocalDateTime.of(2000, 3, 21, 15, 30, 0);
+    int monthVal = 3;
+    // WHEN/THEN
+    then(actual).hasMonthValue(monthVal);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_hasMonthValue_Test.java:54]: inline `monthVal` as it used only once

File: src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_hasMonth_Test.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      .isThrownBy(code)
+      .withMessage("The given Month should not be null");
+  }
+
+  @Test
+  void should_fail_if_actual_is_null() {
+    // GIVEN
+    LocalDateTime actual = null;
+    // WHEN
+    AssertionError assertionError = expectAssertionError(() -> assertThat(actual).hasMonth(LocalDateTime.now().getMonth()));
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_hasMonth_Test.java:49]: use a `Month` value directly, no need to create a `LocalDateTime` just to get its month

File: src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_hasMonth_Test.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    then(assertionError).hasMessage(shouldHaveMonth(actual, month).create());
+  }
+
+  @Test
+  void should_pass_if_actual_is_in_given_month() {
+    // GIVEN
+    LocalDateTime actual = LocalDateTime.of(2022, Month.APRIL, 16, 20, 18, 59);
+    Month month = Month.APRIL;
+    // WHEN/THEN
+    then(actual).hasMonth(month);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_hasMonth_Test.java:71]: inline `month`

**Review by joel-costigliola — CHANGES_REQUESTED** [review]
Thanks @Zhou-Yicheng, first round of code review done, just a few minor comments to address.

File: src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_hasMonth_Test.java (hunk 2 above)
Comment (Zhou-Yicheng) [reply src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_hasMonth_Test.java]: I fix the same style in hasMonthValue_Test too, using a month value directly

File: src/main/java/org/assertj/core/api/AbstractLocalDateTimeAssert.java [hunk 4]
```diff
@@ -742,6 +745,54 @@ public SELF isStrictlyBetween(String startExclusive, String endExclusive) {
...
 
+  /**
+   * Verifies that actual {@link LocalDateTime} is in the given {@link Month}.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // Assertion succeeds:
+   * assertThat(LocalDateTime.of(2022, Month.APRIL, 16, 20, 18, 59)).hasMonth(Month.APRIL);
+   *
+   * // Assertion fails:
+   * assertThat(LocalDateTime.of(2022, Month.APRIL, 16, 20, 18, 59)).hasMonth(Month.May); </code></pre>
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractLocalDateTimeAssert.java:756]: it's `MAY`

**Comment by Zhou-Yicheng** [comment]
Should I update my branch to resolve conflicts?

**Comment by scordio** [comment]
@Zhou-Yicheng yes, please 🙂

**Pull request closed without merging**